swarm/src/behaviour.rs

```diff
@@ -152,97 +152,224 @@ pub trait NetworkBehaviour: 'static {
         vec![]
     }
 
+    /// Informs the behaviour about an event from the [`Swarm`](crate::Swarm).
+    fn on_swarm_event(&mut self, _event: FromSwarm<Self::ConnectionHandler>) {}
+
+    /// Informs the behaviour about an event generated by the [`ConnectionHandler`] dedicated to the
+    /// peer identified by `peer_id`. for the behaviour.
+    ///
+    /// The [`PeerId`] is guaranteed to be in a connected state. In other words,
+    /// [`FromSwarm::ConnectionEstablished`] has previously been received with this [`PeerId`].
+    fn on_connection_handler_event(
+        &mut self,
+        _peer_id: PeerId,
+        _connection_id: ConnectionId,
+        _event: crate::THandlerOutEvent<Self>,
```

Comment:
@thomaseizinger @mxinden regarding https://github.com/libp2p/rust-libp2p/pull/2867/files#r973871955,  doc replaces the alias with the full definition, do you still want to change it?

![image](https://user-images.githubusercontent.com/1204690/195149628-18e0b449-5504-4b75-98c2-deb97e100e84.png)

Reply:
No strong opinion here. Intuition tells me that the full definition is cleaner and easier to understand when reading the code itself. Will let @thomaseizinger decide.

Reply:
I am in favor of the full definition.

Reply:
yeah agreed it's cleaner and easier to understand, updated!  
Btw Thomas, even if the type alias is not public, [Rust seems to be able to point to the aliased one](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d67bb8a886212da10623213eeabc3f80) was there another reason you suggest the type alias be public?

Reply:
Yes. When overriding the callback, users will copy the signature (or IntelliJ / Rust-analyzer will do it for them). If they can't reference the type alias, that is an additional friction point.